elfi/methods/diagnostics.py

```diff
@@ -0,0 +1,281 @@
+                names_ss_MRSSE = names_ss
+                set_ss_2stage = set_ss
+            logger.info('Combination %s shows the MRSSE of %f' % (names_ss, MRSSE_ss))
+        logger.info('\nThe minimum MRSSE of %f was found in %s.' % (MRSSE_min, names_ss_MRSSE))
+        return set_ss_2stage
+
+    def _obtain_accepted_thetas(self, set_ss, n_sim, n_acc):
+        """Obtain the parameters accepted by the ABC-rejection sampling.
+
+        The sampling is performed using the initialised simulator.
+
+        Parameters
+        ----------
+        set_ss : array_like
+            Combination of the summary statistics used in the rejection sampling.
+        n_sim : int
+            Number of the iterations of the rejection sampling.
+        n_acc : int
+            Number of the accepted parameters.
+
+        Returns
+        -------
+        array_like
+            Accepted parameters.
+
+        """
+        # Initialise the distance function.
+        list_ss = []
+        for ss in set_ss:
+            list_ss.append(elfi.Summary(ss, self.simulator))
+        if isinstance(self.fn_distance, str):
+            d = elfi.Distance(self.fn_distance, *list_ss)
+        else:
+            d = elfi.Discrepancy(self.fn_distance, *list_ss)
+
+        # Run the simulations.
+        sampler_rejection = elfi.Rejection(d, batch_size=1, seed=self.seed)
+        result = sampler_rejection.sample(n_acc, n_sim=n_sim)
+
+        # Extract the accepted parameters.
+        thetas_acc = result.samples_array
+        return thetas_acc
+
+    def _calc_entropy(self, thetas_ss, n_acc, k):
+        """Calculate the entropy as described in Nunes & Balding, 2010.
+
+        E = log( pi^(q/2) / gamma(q/2+1) ) - digamma(k) + log(n)
+            + q/n * sum_{i=1}^n( log(R_{i, k}) ), where
+
+        R_{i, k} is the Euclidean distance from the parameter theta_i to
+        its kth nearest neighbour;
+        q is the dimensionality of the parameter; and
+        n is the number of the accepted parameters n_acc in the rejection sampling.
+
+        Parameters
+        ----------
+        thetas_ss : array_like
+            Accepted parameters in the rejection sampling using
+            the combination of summary statistics ss.
+        n_acc : int
+            Number of the accepted parameters.
+        k : int
+            Nearest neighbour to be searched.
+
+        Returns
+        -------
+        float
+            Entropy.
+
+        """
+        q = thetas_ss.shape[1]
+
+        # Calculate the distance to the kth nearest neighbour across all accepted parameters.
+        searcher_knn = cKDTree(thetas_ss)
+        sum_log_dist_knn = 0
+        for theta_ss in thetas_ss:
+            dist_knn = searcher_knn.query(theta_ss, k=k)[0][-1]
+            sum_log_dist_knn += np.log(dist_knn)
+
+        # Calculate the entropy.
+        E = (q / 2) * np.log(np.pi / gamma((q / 2) + 1)) - digamma(k) \
+            + np.log(n_acc) + (q / n_acc) * sum_log_dist_knn
```

Comment:
So the exponent `q/2` was only for the numerator, so you can't take it in front of the combined log. So you'd have to split the log (or keep the exponent, since this really is a drop in the ocean).

Reply:
Sorry! Reverted back to previous version.